src/shared/log-task-chain.c

```diff
@@ -0,0 +1,68 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <string.h>
+
+#include "alloc-util.h"
+#include "format-util.h"
+#include "log-task-chain.h"
+#include "log.h"
+#include "process-util.h"
+
+void log_task_chain_msg(sd_bus_message *message, const char *special_action_string) {
+        _cleanup_(sd_bus_creds_unrefp) sd_bus_creds *creds = NULL;
+        pid_t pid;
+
+        assert(message);
+        assert(special_action_string);
+
+        if (sd_bus_query_sender_creds(message, SD_BUS_CREDS_PID|SD_BUS_CREDS_AUGMENT|SD_BUS_CREDS_COMM, &creds) < 0)
+                return;
+
+        /* We need at least the PID, otherwise there's nothing to log, the rest is optional */
+        if (sd_bus_creds_get_pid(creds, &pid) < 0)
+                return;
+
+        log_task_chain_pid(pid, special_action_string);
+}
+
+void log_task_chain_pid(pid_t pid, const char *special_action_string) {
+        pid_t ppid;
+        uid_t uid;
+
+        int saved_facility = log_get_facility();
+        log_set_facility(LOG_AUTHPRIV);
```

Comment:
Drop this as well, we don't do this either

Reply:
OK, but will use `log_info` instead of `log_debug` requested earlier.